test/integration/irohad_test.cpp

```diff
@@ -0,0 +1,56 @@
+/**
+ * Copyright Soramitsu Co., Ltd. 2018 All Rights Reserved.
+ * http://soramitsu.co.jp
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <boost/process.hpp>
+#include "gtest/gtest.h"
+
+using namespace boost::process;
+using namespace std::chrono_literals;
+
+const std::string path_irohad(PATHIROHAD);
+const std::string irohad_executable = path_irohad + "/irohad";
+const std::string path_example = path_irohad + "/../../example";
+const std::string path_config = path_example + "/config.sample";
+const std::string path_genesis = path_example + "/genesis.block";
+const std::string path_keypair = path_example + "/node0";
+const std::string params = "--config " + path_config + " --genesis_block "
+    + path_genesis + " --keypair_name " + path_keypair;
+const auto timeout = 1s;
+
+/*
+ * @given path to irohad executable
+ * @when run irohad without any parameters
+ * @then irohad should not start
+ */
+TEST(IrohadTest, RunIrohadWithoutArgs) {
+  child c(irohad_executable);
+  c.wait_for(timeout);
```

Comment:
I'm pretty sure there's a return code, it's better check it than depend on a timeout

Reply:
I should wait some time anyway because child starts async.

Reply:
[join](http://www.boost.org/doc/libs/1_64_0/doc/html/boost/process/child.html#idp24620736-bb)?

Reply:
But wait_for is what I'm actually use :)

Reply:
I meant with the error_code, though not really sure that it stops after process exiting

Reply:
Hmm, from error_code it looks like no actual error exists (Undefined error, code 0). I improved this piece of code a bit anyway.